beetsplug/zero.py

```diff
@@ -41,26 +41,50 @@ def __init__(self):
 
         self.config.add({
             'fields': [],
-            'update_database': False,
+            'keep_fields': [],
+            'update_database': False
         })
 
         self.patterns = {}
         self.warned = False
 
-        for field in self.config['fields'].as_str_seq():
-            if field in ('id', 'path', 'album_id'):
-                self._log.warn(u'field \'{0}\' ignored, zeroing '
-                               u'it would be dangerous', field)
-                continue
+        if self.config['fields']:
+            self.validate_config('fields')
+            for field in self.config['fields'].as_str_seq():
+                self.write_patterns(field)
+
+        elif self.config['keep_fields']:
+            self.validate_config('keep_fields')
+
+            for field in MediaFile.fields():
+                if field in self.config['keep_fields'].as_str_seq():
+                    continue
+                self.write_patterns(field)
+
+            # These fields should be preserved
+            for key in ('id', 'path', 'album_id'):
+                if key in self.patterns:
+                    del self.patterns[key]
+
+    def validate_config(self, mode):
+        """Check if fields written in config are correct."""
+        if self.config['fields'] and self.config['keep_fields']:
+            self._log.warn(u'cannot blacklist and whitelist at the same time')
+        for field in self.config[mode].as_str_seq():
             if field not in MediaFile.fields():
                 self._log.error(u'invalid field: {0}', field)
                 continue
+            if mode == 'fields' and field in ('id', 'path', 'album_id'):
+                self._log.warn(u'field \'{0}\' ignored, zeroing '
+                               u'it would be dangerous', field)
+                continue
 
-            try:
-                self.patterns[field] = self.config[field].as_str_seq()
-            except confit.NotFoundError:
-                # Matches everything
-                self.patterns[field] = True
+    def write_patterns(self, field):
```

Comment:
May I suggest the slightly more descriptive name `set_pattern`? And a brief docstring, like this, would help too:

> Set a field in `self.patterns` to a string list corresponding to the configuration, or True if the field has no specific configuration.